cmd/get-measurements/main.go

```diff
@@ -120,32 +141,41 @@ func runClient(cCtx *cli.Context) (err error) {
 	certs := resp.TLS.PeerCertificates
 
 	// Extract the aTLS variant and measurements from the TLS connection
-	// certs := conn.ConnectionState().PeerCertificates
 	atlsVariant, extractedMeasurements, err := proxy.GetMeasurementsFromTLS(certs, []asn1.ObjectIdentifier{variant.AzureTDX{}.OID()})
 	if err != nil {
 		log.Error("Error in getMeasurementsFromTLS", "err", err)
 		return err
 	}
 
-	measurementsInHeaderFormat := make(map[uint32]string, len(extractedMeasurements))
-	for pcr, value := range extractedMeasurements {
-		measurementsInHeaderFormat[pcr] = hex.EncodeToString(value)
+	printableMeasurements := make(map[uint32]string)
+	for k, v := range extractedMeasurements {
+		printableMeasurements[k] = fmt.Sprintf("%x", v)
 	}
 
-	marshaledPcrs, err := json.MarshalIndent(measurementsInHeaderFormat, "", "    ")
+	marshaledPcrs, err := json.MarshalIndent(printableMeasurements, "", "    ")
 	if err != nil {
 		return errors.New("could not marshal measurement extracted from tls extension")
 	}
 
 	log.Info("Variant: " + atlsVariant.String())
-	log.Info(fmt.Sprintf("Measurements for %s with %d entries:", atlsVariant.String(), len(measurementsInHeaderFormat)))
+	log.Info(fmt.Sprintf("Measurements for %s with %d entries:", atlsVariant.String(), len(extractedMeasurements)))
 	fmt.Println(string(marshaledPcrs))
 	if outMeasurements != "" {
-		if err := os.WriteFile(outMeasurements, marshaledPcrs, 0644); err != nil {
+		if err := os.WriteFile(outMeasurements, marshaledPcrs, 0o644); err != nil {
 			return err
 		}
 	}
 
+	// Compare against expected measurements
+	if expectedMeasurements != nil {
+		found, foundMeasurement := expectedMeasurements.Contains(extractedMeasurements)
+		if found {
+			log.With("matchedMeasurements", foundMeasurement.MeasurementID).Info("Measurements match expected measurements ✅")
+		} else {
+			log.Error("Measurements do not match expected measurements! ❌")
```

Comment:
Suggestion: I would make the error message more verbose. For example, "The fetched %instance% measurements do not match any entry of the provided expected measurements list".

Reply:
since the content is basically the same, i think short is better. at least to start with